core/Tracker/Model.php

```diff
@@ -422,7 +422,7 @@ public function hasVisit($idSite, $idVisit)
         $bindSql = array($idVisit);
 
         $val = $this->getDb()->fetchOne($sql, $bindSql);
-        return $val == $idSite;
+        return $val === $idSite;
```

Comment:
Database query might return a string, while the value given to that method could actually be string or int I guess
```suggestion
        return (string) $val === (string) $idSite;
```